certification/internal/shell/has_v2_related_image_schema_version.go

```diff
@@ -0,0 +1,230 @@
+	if err != nil {
+		log.Error(err)
+		return nil, fmt.Errorf("%w: %s", errors.ErrDeterminingRelatedImageSchemaVers, err)
+	}
+
+	return imageSchemaVersionMap, nil
+}
+
+// extractBundleFromImage will extract the bundle contents from the specified image, and return the
+// path to the bundle on the filesystem. This is done by creating the container, copying the contents
+// out of the container, and then cleaning up the container.
+func (p *RelatedImagesAreSchemaVersion2Check) extractBundleFromImage(image string) (string, error) {
+	localBundlePath := "extracted-bundle"
+
+	// create the container. Bundle containers are non-runnable, so we need to
+	// add a dummy entrypoint so that we can create the container.
+	createReport, err := podmanEngine.Create(image, &cli.PodmanCreateOptions{Entrypoint: "false"})
+	if err != nil {
+		log.Error("stdout: ", createReport.Stdout)
+		log.Error("stderr: ", createReport.Stderr)
+		return "", err
+	}
+
+	defer func() {
+		deleteReport, err := podmanEngine.Remove(createReport.ContainerID)
+		if err != nil {
+			log.Warn("a non-fatal error occurred while trying to remove container: ", err)
+			log.Warn("stdout: ", deleteReport.Stdout)
+			log.Warn("stderr: ", deleteReport.Stderr)
+		}
+	}()
+
+	copyReport, err := podmanEngine.CopyFrom(createReport.ContainerID, "/", localBundlePath)
+	if err != nil {
+		log.Error("stdout: ", copyReport.Stdout)
+		log.Error("stderr: ", copyReport.Stderr)
+		return "", err
+	}
+
+	return localBundlePath, nil
+}
+
+// readBundle will accept the manifests directory where a bundle is expected to live,
+// and walks the directory to find all bundle assets.
+func (p *RelatedImagesAreSchemaVersion2Check) readBundle(manifestsDir string) (*operatorsv1alpha1.ClusterServiceVersion, error) {
+	bundle, err := manifests.GetBundleFromDir(manifestsDir)
+	if err != nil {
+		return nil, fmt.Errorf("%w: %s", errors.ErrDeterminingRelatedImageSchemaVers, err)
+	}
+
+	return bundle.CSV, nil
+}
+
+// getRelatedImagesForCSV will return a slice of strings containing the images found in the relatedImages field of
+// the input ClusterServiceVersion.
+func (p *RelatedImagesAreSchemaVersion2Check) getRelatedImagesForCSV(csv *operatorsv1alpha1.ClusterServiceVersion) ([]string, error) {
+	relatedImages := csv.Spec.RelatedImages
+
+	// no related images == nothing to check
+	if len(relatedImages) == 0 {
+		return nil, nil
+	}
+
+	imageStrings := make([]string, len(relatedImages))
+	for i, relatedImage := range relatedImages {
+		imageStrings[i] = relatedImage.Image
+	}
+
+	return imageStrings, nil
+}
+
+// inspectSchemaVersionForImages will inspect each input image using skopeo and returns a map of each image
+// with its corresponding schemaVersion. It is expeted that len(images) > 0.
+func (p *RelatedImagesAreSchemaVersion2Check) inspectSchemaVersionForImage(images []string) (map[string]int, error) {
+	inspectOpts := cli.SkopeoInspectOptions{Raw: true}
+
+	mapping := map[string]int{}
```

Comment:
How about imageToSchemaVersion? This makes it easier to understand what is the purpose of this variable

Reply:
Fixed!